src/jdmtool/main.py

```diff
@@ -104,18 +107,16 @@ def wrapper(dev: SkyboundDevice, *args, **kwargs):
         dev.before_read()
 
         # TODO: Figure out the actual meaning of the iid and the "unknown" value.
-        iid = dev.get_iid()
-        if iid == 0x01004100:
-            # 16MB WAAS card
-            print("Detected data card: 16MB WAAS")
-            dev.set_memory_layout(SkyboundDevice.MEMORY_LAYOUT_16MB)
-        elif iid == 0x0100ad00:
-            # 4MB non-WAAS card
-            print("Detected data card: 4MB non-WAAS")
-            dev.set_memory_layout(SkyboundDevice.MEMORY_LAYOUT_4MB)
+        card = dev.get_card()
+
+        log.debug(f"Detected card: {card}")
+
+        if card:
+            print(f"Detected data card: {card}")
+            dev.set_memory_layout(card.memory_layout)
         else:
             raise SkyboundException(
-                f"Unknown data card IID: 0x{iid:08x} (possibly 8MB non-WAAS?). Please file a bug!"
+                f"Unknown data card IID: 0x{dev.get_iid():08x} (possibly 8MB non-WAAS?). Please file a bug!"
```

Comment:
`get_iid()` actually does USB I/O (ok, probably not obvious given the `get_` in its name), so it really should be cached somewhere.